app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/catalog/utils/ViewComponents.kt

```diff
@@ -0,0 +1,137 @@
+import androidx.constraintlayout.compose.ConstraintLayout
+import androidx.constraintlayout.compose.ConstraintSet
+import com.microsoft.device.samples.dualscreenexperience.R
+
+const val CONTENT_ID = "tableOfContents"
+const val BOTTOM_PAGE_NUMBER_ID = "bottomPageNumber"
+
+@Composable
+fun PageLayout(
+    modifier: Modifier = Modifier,
+    pageNumber: Int,
+    maxPageNumber: Int,
+    content: @Composable () -> Unit
+) {
+    val constraintSet = getMainConstraintSet()
+
+    ConstraintLayout(constraintSet = constraintSet, modifier = modifier) {
+        Box(
+            modifier = Modifier
+                .fillMaxHeight()
+                .padding(bottom = dimensionResource(id = R.dimen.catalog_margin_normal))
+                .layoutId(CONTENT_ID)
+        ) {
+            content()
+        }
+
+        BottomPageNumber(
+            modifier = Modifier.layoutId(BOTTOM_PAGE_NUMBER_ID),
+            text = stringResource(
+                id = R.string.catalog_page_no, pageNumber, maxPageNumber
+            )
+        )
+    }
+}
+
+@Composable
+private fun getMainConstraintSet() = ConstraintSet {
+    val contentRef = createRefFor(CONTENT_ID)
+    val bottomPageNumberRef = createRefFor(BOTTOM_PAGE_NUMBER_ID)
+
+    constrain(contentRef) {
+        linkTo(start = parent.start, end = parent.end)
+        linkTo(top = parent.top, bottom = bottomPageNumberRef.top)
+    }
+
+    constrain(bottomPageNumberRef) {
+        start.linkTo(parent.start)
+        bottom.linkTo(parent.bottom)
+    }
+}
+
+@Composable
+fun BottomPageNumber(modifier: Modifier = Modifier, text: String) {
+    Text(
+        text = text,
+        style = TextStyle(
+            color = MaterialTheme.colors.onSurface,
+            fontFamily = FontFamily(Font(R.font.dmsans_regular)),
+            lineHeight = fontDimensionResource(id = R.dimen.text_size_4),
+            fontSize = fontDimensionResource(id = R.dimen.text_size_12),
+            textAlign = TextAlign.Start
+        ),
+        modifier = modifier.padding(
+            start = dimensionResource(id = R.dimen.catalog_horizontal_margin),
+            end = dimensionResource(id = R.dimen.catalog_horizontal_margin),
+            bottom = dimensionResource(id = R.dimen.catalog_margin_small)
+        )
+    )
+}
+
+@Composable
+fun TextDescription(
+    modifier: Modifier = Modifier,
+    text: String,
+    fontSize: TextUnit = fontDimensionResource(id = R.dimen.text_size_12),
+    lineHeight: TextUnit = fontDimensionResource(id = R.dimen.text_size_20),
+    contentDescription: String,
+    textAlign: TextAlign = TextAlign.Start
+) {
+    Text(
+        modifier = modifier.contentDescription(contentDescription),
+        text = text,
+        lineHeight = lineHeight,
+        textAlign = textAlign,
+        style = TextStyle(
+            color = MaterialTheme.colors.onSurface,
+            fontSize = fontSize,
+            fontFamily = FontFamily(Font(R.font.dmsans_regular))
+        )
+    )
+}
```

Comment:
Related to my other comment about not needing to use copy, you also have the option here to just pass in a themed text style to cover the font size/family - it's up to you which way to use, but like Joy mentioned earlier it might be easier to change the text size of multiple composables later on using the theme!

Also, have you had issues with lineHeight default value? I noticed in a few places you specify font size but not line height, and I wasn't sure if that would affect the appearance if the font size is bigger than the line height for some case
```suggestion
@Composable
fun TextDescription(
    modifier: Modifier = Modifier,
    text: String,
    textStyle: TextStyle,
    lineHeight: TextUnit = fontDimensionResource(id = R.dimen.text_size_20),
    contentDescription: String,
    textAlign: TextAlign = TextAlign.Start
) {
    Text(
        modifier = modifier.contentDescription(contentDescription),
        text = text,
        lineHeight = lineHeight,
        textAlign = textAlign,
        color = MaterialTheme.colors.onSurface,
        style = textStyle
    )
}
```